syntax = "proto3";

import "tensorflow/core/framework/tensor.proto";

package server;

service Server {
    // (Method definitions not shown)
    rpc serveModel(ModelInput) returns (ModelOutput) {}
}

message ModelInput {
    int32 requestId = 1;

    string modelName = 2;
    string layerName = 3;
    tensorflow.TensorProto tensor = 4;
}

message ModelOutput {
    bool hasValue = 1;
    map<string, tensorflow.TensorProto> result = 2;
}